core/services/job_runner.go

```diff
@@ -267,6 +268,19 @@ func executeRun(run *models.JobRun, store *store.Store) error {
 
 	if run.Status.Finished() {
 		run.SetFinishedAt()
+		if run.Result.Amount != nil {
+			existingJob, err := store.FindJob(run.JobSpecID)
```

Comment:
Rather than doing this all in code, perhaps we could do it when writing to the database. `SaveRun` or `UpdateRun` could `SUM` up all of the LINK for the job. With that said, right now we've treated `JobSpec` as immutable, and this would introduce the first mutating value.

You could potentially put this on a whole new table and store it there. There are ideas. I think in general you have the right idea in that we want to save the total on write, rather than calculating it on every read.